app/controllers/reacts_controller.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+# react emotion
+class ReactsController < ApplicationController
+  before_action :logged_in_user, only: %i[create destroy]
+  before_action :set_micropost, only: %i[create destroy]
+
+  def create
+    return render json: { success: false, status: 'error' } if React.find_by(user_id: params[:react][:user_id], micropost_id: params[:react][:micropost_id])
+
+    @react = @micropost.reacts.new(react_params)
+    @react.user_id = params[:react][:user_id]
+    if @react.save
+      html_content = render_to_string(partial: 'reacts/remove_action', locals: { emotion: @react.action, micropost: @micropost }).squish
+      html_total_react = render_to_string(partial: 'reacts/total_react', locals: { micropost: @micropost }).squish
+      render json: { success: true, emotion: @react, status: 'success', html_content:, html_total_react: }
+    else
+      render json: { success: false, status: 'error' }
+    end
+  end
+
```

Comment:
What if user want to change reaction? 
With the current logic user need to delete react first and then react, add Logic to allow user to update current reaction to another reaction

Reply:
fixed